master/internal/trial.go

```diff
@@ -206,6 +210,7 @@ func (t *trial) maybeAllocateTask(ctx *actor.Context) error {
 	}
 
 	ctx.Log().Info("decided to allocate trial")
+	t.runID += 1
```

Comment:
question: is this a separate bugfix?

Reply:
yes, but this PR is dependent on it

Reply:
great. if it's user-facing we might want a small mention release notes.